process-engine/consumer_api_core
# ⚡️ ✨ Add lane and name to flow node instance

## Changes

1. Use the value of `FlowNodeInstance.flowNodeLane` to perform claim checks for task accessibility 
    - Includes UserTasks, ManualTasks, Events and EmptyActivities
2. Perform task filtering and pagination prior to conversion
3. Remove retrieval of a tasks's ProcessModel from the ManualTaskService and EmptyActivityService
    - This was not entirely possible for the other services, because Events and UserTasks require a lot of information from their models that cannot be added to a FlowNodeInstance (the UserTask's FormField Configs, for example)
    - The services will benefit from increased performance, because they now only have to perform a fraction of the `getProcessModel` queries they did before
4. Account for `SuperAdmin` claim during all queries, which greatly reduces the number of claim checks performed, if the requesting user actually is a super admin
5. When a user is not allowed to see any suspended task, he will now receive an empty array, instead of a 403. 
    - This fixes a possible issue, where a user might get more information about existing tasks, than he is supposed to 
    - i.e.: 403 says "there is something there, I just can't see it"; an empty Array suggests "there is nothing there to see"

## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/427

PR: #109

## How to test the changes

- Run various requests for retrieving ManualTasks, UserTasks, Events and/or EmptyActivities
- The more tasks you retrieve, the greater the performance boost you should be able to notice
